File: solutions/python/allergies/1/allergies.py
from asyncio import selector_events
import math
class Allergies:
    Allergens = {
        "eggs":0,
        "peanuts":1,
        "shellfish":2,
        "strawberries":3,
        "tomatoes":4,
        "chocolate":5,
        "pollen":6,
        "cats":7
    }
    allergens_arr = ["eggs","peanuts","shellfish","strawberries","tomatoes","chocolate","pollen","cats"]
    def __init__(self, score):
        self.score = score % 256;

    def allergic_to(self, item):
        score_arr = [0]*8
        item_arr = [0]*8
        score_bin = bin(self.score)[2:]
        item_score = 2**(self.Allergens[item])
        item_bin= bin(item_score)[2:]
        i = -1 
        for it in item_bin[::-1]:
            item_arr[i] = 1 if it == '1' else 0
            i = i - 1
        j = - 1
        for sb in score_bin[::-1]:
            score_arr[j] = 1 if sb == '1' else 0
            j = j - 1
        for a in range(len(item_arr)):
            if item_arr[a] == 1:
                if score_arr[a] == 0:
                    return False
                else:
                    return True
    @property
    def lst(self):
        score_arr = [0]*8
        score_bin = bin(self.score)[2:]
        j = - 1
        for sb in score_bin[::-1]:
            score_arr[j] = 1 if sb == '1' else 0
            j = j - 1
        res = []
        for idx,ele in enumerate(score_arr):
            if ele == 1:
                res.append(self.allergens_arr[-(idx+1)])
        return res

# Allergies(2).lst

File: solutions/c/allergies/1/allergies.c
#include "allergies.h"

bool is_allergic_to(allergen_t allergen, int score) {
  score %= 256;
  return (score & (1 << allergen)) != 0;
}

allergen_list_t get_allergens(int score) {
  allergen_list_t list;
  list.count = 0;

  score %= 256;

  for (int i = 0; i < ALLERGEN_COUNT; i++) {
    list.allergens[i] = is_allergic_to((allergen_t)i, score);

    if (list.allergens[i]) {
      list.count++;
    }
  }

  return list;
}